File: framework/lib/effect.dart
part of uxml;

/**
  * Provides base class for chrome effects.
  *
  * @author:ferhat@ (Ferhat Buyukkokten)
  */
class Effect extends UxmlElement{
  static int _idGen = 0;
  int _signature;

  List<Action> _actions;
  Object property;
  Object source;
  Object targetElement;
  Object value;

  Effect() : super() {
    _signature = _idGen++;
    _actions = <Action>[];
  }

  /**
   * Adds an action to perform when effect is activated.
   */
  void addAction(Action action) {
    _actions.add(action);
  }

   /**
    * Returns id for effect that is shared across clones.
    */
  String get id => _signature.toString();

  /**
   * Returns list of actions to execute for effect.
   */
  List<Action> get actions => _actions;

  Effect clone() {
    Effect newEffect = new Effect();
    newEffect._actions = _actions;
    newEffect.property = property;
    newEffect.source = source;
    newEffect.targetElement = targetElement;
    newEffect.value = value;
    newEffect._signature = _signature;
    return newEffect;
  }
}

File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/effects/Effect.java
package com.hello.uxml.tools.framework.effects;

import com.hello.uxml.tools.framework.ContentNode;
import com.hello.uxml.tools.framework.PropertyDefinition;
import com.hello.uxml.tools.framework.UxmlElement;

import java.util.ArrayList;
import java.util.List;

/**
 * Holds behaviour information for chrome instances.
 *
 * @author ferhat@(Ferhat Buyukkokten)
 */
public class Effect extends UxmlElement {
  private List<Action> actions = new ArrayList<Action>();
  private PropertyDefinition property;
  private Object value;
  private Object source;
  private UxmlElement targetElementValue;

  /**
   * Adds an action to effect.
   */
  @ContentNode
  public void addAction(Action action) {
    actions.add(action);
  }

  /**
   * Sets or returns property that triggers actions.
   */
  public void setProperty(PropertyDefinition property) {
    this.property = property;
  }

  public PropertyDefinition getProperty() {
    return property;
  }

  /**
   * Sets or returns source of property that triggers actions.
   */
  public void setSource(Object source) {
    this.source = source;
  }

  public Object getSource() {
    return source;
  }

  /**
   * Sets or returns targetElement for action.
   */
  public void setTargetElement(UxmlElement target) {
    this.targetElementValue = target;
  }

  public UxmlElement getTargetElement() {
    return targetElementValue;
  }

  /**
   * Sets or returns value of property that triggers actions.
   */
  public void setValue(Object value) {
    this.value = value;
  }

  public Object getValue() {
    return value;
  }

  /**
   * Returns list of actions to execute.
   */
  public List<Action> getActions() {
    return actions;
  }
}
